src/main/kotlin/io/spinnaker/bumpdeps/NonDestructivePropertiesEditor.kt

```diff
@@ -0,0 +1,48 @@
+package io.spinnaker.bumpdeps
+
+import java.io.FileWriter
+import java.nio.charset.Charset
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.regex.Pattern
+
+class NonDestructivePropertiesEditor(internal val propertiesFile: Path) {
```

Comment:
I'd just inline `saveResult` into `Main.kt` and then make `propertiesFile` be a parameter of the first `updateProperty` method... I don't think having it as a property of this class really adds anything...

Reply:
The other thing it is uses for is the `updateProperties(key, value)` (`updateProperty(lines, key, value)` was mostly for testability more than anything)

I'll fix the saveResult bit but keep the ownership of file io inside this class unless you feel strongly the other direction

Reply:
k, fair enough, thanks!